File: core/x/javaapi/codegen/src/test/resources/golden/expected/com/example/grts/Auditable.java
package com.example.grts;

import viaduct.java.api.globalid.GlobalID;
import viaduct.java.api.reflect.CompositeField;
import viaduct.java.api.reflect.Field;
import viaduct.java.api.reflect.Type;
import viaduct.java.api.reflect.TypeFields;
import viaduct.java.api.types.GraphQLInterface;
import viaduct.java.api.types.NodeCompositeOutput;
import java.time.Instant;
import java.time.LocalDate;
import java.time.OffsetTime;
import java.util.List;

@SuppressWarnings("MissingOverride")
public interface Auditable extends NodeCompositeOutput, Node, Timestamped {

    Type<Auditable> Reflection = Type.ofClass(Auditable.class);

    final class Fields implements TypeFields<Auditable> {
        private Fields() {}

        public static final Field<Auditable> __typename =
                Field.of("__typename", Reflection);
                public static final Field<Auditable> id =
                                Field.of("id", Reflection);

                public static final Field<Auditable> createdAt =
                                Field.of("createdAt", Reflection);

                public static final Field<Auditable> updatedAt =
                                Field.of("updatedAt", Reflection);

                public static final Field<Auditable> auditTrail =
                                Field.of("auditTrail", Reflection);

    }

    GlobalID<? extends Auditable> getIdOrThrow(String alias);
    GlobalID<? extends Auditable> getIdOrThrow();
    GlobalID<? extends Auditable> getId(String alias);
    GlobalID<? extends Auditable> getId();
    String getCreatedAtOrThrow(String alias);
    String getCreatedAtOrThrow();
    String getCreatedAt(String alias);
    String getCreatedAt();
    String getUpdatedAtOrThrow(String alias);
    String getUpdatedAtOrThrow();
    String getUpdatedAt(String alias);
    String getUpdatedAt();
    List<String> getAuditTrailOrThrow(String alias);
    List<String> getAuditTrailOrThrow();
    List<String> getAuditTrail(String alias);
    List<String> getAuditTrail();

}

File: core/tenant/codegen/src/test/resources/golden/expected/grts/Auditable.kt
@file:Suppress("warnings")

package viaduct.api.grts

interface Auditable : viaduct.api.types.Interface, viaduct.api.types.NodeCompositeOutput, viaduct.api.grts.Node, viaduct.api.grts.Timestamped {
          fun getAuditTrailOrThrow(alias: String?): kotlin.collections.List<kotlin.String>?
          fun getAuditTrailOrThrow(): kotlin.collections.List<kotlin.String>?
          fun getAuditTrail(alias: String?): kotlin.collections.List<kotlin.String>?
          fun getAuditTrail(): kotlin.collections.List<kotlin.String>?


    @OptIn(viaduct.apiannotations.InternalApi::class)
    object Reflection : viaduct.api.reflect.Type<viaduct.api.grts.Auditable> {
        override final val name = "Auditable"
        override final val kcls = viaduct.api.grts.Auditable::class
    }
    object Fields : viaduct.api.reflect.TypeFields<viaduct.api.grts.Auditable> {
            final val __typename: viaduct.api.reflect.Field<viaduct.api.grts.Auditable> =
                viaduct.api.internal.FieldImpl("__typename", viaduct.api.grts.Auditable.Reflection)

            final val id: viaduct.api.reflect.Field<viaduct.api.grts.Auditable> =
                viaduct.api.internal.FieldImpl("id", viaduct.api.grts.Auditable.Reflection)

            final val createdAt: viaduct.api.reflect.Field<viaduct.api.grts.Auditable> =
                viaduct.api.internal.FieldImpl("createdAt", viaduct.api.grts.Auditable.Reflection)

            final val updatedAt: viaduct.api.reflect.Field<viaduct.api.grts.Auditable> =
                viaduct.api.internal.FieldImpl("updatedAt", viaduct.api.grts.Auditable.Reflection)

            final val auditTrail: viaduct.api.reflect.Field<viaduct.api.grts.Auditable> =
                viaduct.api.internal.FieldImpl("auditTrail", viaduct.api.grts.Auditable.Reflection)

    }

}